llvm/lib/SYCLLowerIR/ESIMD/LowerESIMDKernelAttrs.cpp

```diff
@@ -12,14 +12,82 @@
 #include "llvm/SYCLLowerIR/ESIMD/LowerESIMD.h"
 #include "llvm/SYCLLowerIR/SYCLUtils.h"
 
-#include "llvm/IR/Module.h"
-#include "llvm/Pass.h"
-
 #define DEBUG_TYPE "LowerESIMDKernelAttrs"
 
 using namespace llvm;
 
 namespace llvm {
+
+// Checks if Call Instruction corresponds to InvokeSimd function call.
+bool isInvokeSimdBuiltinCall(const CallInst *CI) {
+  Function *F = CI->getCalledFunction();
+
+  if (F && F->getName().startswith(esimd::INVOKE_SIMD_PREF)) {
+    return true;
+  }
+  return false;
+}
+
+// Checks the use of a function address being stored in a memory.
+// Returns false if the function address is used as an argument for
+// invoke_simd function call, true otherwise.
+bool checkFunctionAddressUse(const Value *address) {
+  if (address == nullptr) {
+    return true;
+  }
```

Comment:
Note: [LLVM Coding Standard](https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements) says don’t use braces on simple single-statement bodies of if/else/loop statements

Reply:
I always thought this is not strict - `we prefer to omit the braces to avoid unnecessary line noise`.